pkg/apis/query/v0alpha1/query.go

```diff
@@ -0,0 +1,212 @@
+package v0alpha1
+
+import (
+	"encoding/json"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Generic query request with shared time across all values
+// Copied from: https://github.com/grafana/grafana/blob/main/pkg/api/dtos/models.go#L62
+type GenericQueryRequest struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// From Start time in epoch timestamps in milliseconds or relative using Grafana time units.
+	// example: now-1h
+	From string `json:"from,omitempty"`
+
+	// To End time in epoch timestamps in milliseconds or relative using Grafana time units.
+	// example: now
+	To string `json:"to,omitempty"`
+
+	// queries.refId – Specifies an identifier of the query. Is optional and default to “A”.
+	// queries.datasourceId – Specifies the data source to be queried. Each query in the request must have an unique datasourceId.
+	// queries.maxDataPoints - Species maximum amount of data points that dashboard panel can render. Is optional and default to 100.
+	// queries.intervalMs - Specifies the time interval in milliseconds of time series. Is optional and defaults to 1000.
+	// required: true
+	// example: [ { "refId": "A", "intervalMs": 86400000, "maxDataPoints": 1092, "datasource":{ "uid":"PD8C576611E62080A" }, "rawSql": "SELECT 1 as valueOne, 2 as valueTwo", "format": "table" } ]
+	Queries []GenericDataQuery `json:"queries"`
+
+	// required: false
+	Debug bool `json:"debug,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type QueryResults struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// The response data
+	Results map[string]QueryResult `json:"results"`
+}
+
+// Result for a single query
+type QueryResult struct {
+	Frames []string `json:"frames"` // TODO... actually DataFrame!!!
+	Status int64    `json:"status,omitempty"`
+}
+
+type DataSourceRef struct {
+	// The datasource plugin type
+	Type string `json:"type"`
+
+	// Datasource UID
+	UID string `json:"uid"`
+}
+
+type GenericDataQuery struct {
+	// RefID is the unique identifier of the query, set by the frontend call.
+	RefID string `json:"refId"`
+
+	// TimeRange represents the query range
+	// NOTE: unlike generic /ds/query, we can now send explicit time values in each query
+	TimeRange *TimeRange `json:"timeRange,omitempty"`
```

Comment:
^^ this is new.   Currently it exists on raw backend queries, but no in the /ds/query endpoint